cmd/ipfs-cluster-service/main.go

```diff
@@ -304,6 +332,18 @@ remove the %s file first and clean any Raft state.
 					checkErr("saving "+DefaultIdentityFile, err)
 					out("new identity written to %s\n", identityPath)
 				}
+				if peersOpt != "" {
+					_, _, cfgs := makeAndLoadConfigs()
```

Comment:
you have a `cfgs` object already above with all the default things you need, why creating it again by forcing a re-read of all the config file?

In fact, this block can be moved up to the place the peers addresses and handled.

Reply:
`cfgs` object that we have already would result `cfgs.clusterCfg.GetPeerstorePath()` in an empty string (because clusterConfig.BaseDir would be empty)

clusterConfig.BaseDir will be set in cfgMgr.LoadJSON

So, once the json file is saved and then when we load the configs again, `cfgs.clusterCfg.GetPeerstorePath()` would return the right path.

`makeAndLoadConfigs` also loads identity, so I think it is safe to keep it after we have save identity as well.

Reply:
then do a `cfgs.clusterCfg.SetBaseDir(filepath.Dir(configPath))`. Re-reading the configuration file and the identity from disk and re-parsing it is an overkill operation that you don't need.